a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -129,9 +129,24 @@ def execute(self, loadbalancer, vthunder, added_ports):
 class AllowL2DSR(VThunderBaseTask):
     """Task to add wildcat address in allowed_address_pair for L2DSR"""
 
-    def execute(self, subnet, amphora):
-        if CONF.vthunder.l2dsr_support:
-            self.network_driver.allow_use_any_source_ip_on_egress(subnet.network_id, amphora)
+    def execute(self, subnet, amphora, flavor_data=None):
+        if flavor_data:
+            deployment = flavor_data.get('deployment')
+            if deployment and deployment['dsr_type'] == "l2dsr_transparent":
+                for amp in amphora:
+                    self.network_driver.allow_use_any_source_ip_on_egress(subnet.network_id, amp)
+
```

Comment:
can we have revert() function for this task?

Reply:
Added revert() function